libcontainer/configs/cgroup.go

```diff
@@ -57,6 +57,11 @@ type Cgroup struct {
 	// MEM to use
 	CpusetMems string `json:"cpuset_mems"`
 
+	// Process limit; set to `0' to disable limit.
+	// While technically `0' is a valid PID limit, it does not make sense in the
+	// context of a container -- it is identical to a limit of `1'.
```

Comment:
@yangdongsheng The maintainer felt that magical values meaning max (such as `0`) don't match the rest of the cgroup API. Also, using a value like `0` as sentinel value would've added extra checks to the hierarchical limit checking that weren't necessary (the way that it works is to not allow a user to set a numerical value that is identical to an unlimited value -- because that would be confusing). By the way, the default way which all of the limiting features in runc are the same as I've described. If the default value is set for a limit, then the limit isn't changed from the cgroup default (to stop missing cgroups from causing the container to fail starting if it hasn't specified a preference for an unimplemented limit). Trying to say that "0 actually means something" is counter to the rest of the runc configuration style.